### PR TITLE
Refactor Lexer.combineComment to be simpler and accept D slices

### PR DESCRIPTION
**Pull request by Geod24**
Better reviewed commit-by-commit, with whitespace diff off.

### CODE REVIEW

**Comment by WalterBright** [comment]
Seems to me it would be better to change `lineComment` and `blockComment` to D arrays and do away with all the `toDString` calls.

**Comment by Geod24** [comment]
> Seems to me it would be better to change lineComment and blockComment to D arrays and do away with all the toDString calls.

I actually have a commit to do that but wanted to keep the PR small.

**Comment by WalterBright** [comment]
I suspect that would be smaller than this PR.

**Comment by Geod24** [comment]
Done

File: src/dmd/tokens.d [hunk 1]
```diff
@@ -429,8 +429,8 @@ extern (C++) struct Token
     Loc loc;
     const(char)* ptr; // pointer to first character of this token within buffer
     TOK value;
-    const(char)* blockComment; // doc comment string prior to this token
-    const(char)* lineComment; // doc comment for previous token
+    const(char)[] blockComment; // doc comment string prior to this token
+    const(char)[] lineComment; // doc comment for previous token
```
Comment (kinke) [inline src/dmd/tokens.d]: tokens.h too pls.

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**